flink/src/test/java/org/apache/iceberg/flink/TestFlinkTableSource.java

```diff
@@ -75,32 +95,328 @@ public void clean() {
+    assertFalse("explain should not contains FilterPushDown", explainNotInNull.contains(expectedFilterPushDownExplain));
+  }
+
+  @Test
+  public void testFilterPushDownIsNotNull() {
+    String sqlNotNull = String.format("SELECT * FROM %s WHERE data IS NOT NULL", TABLE_NAME);
+    String explainNotNull = getTableEnv().explainSql(sqlNotNull);
+    assertTrue("explain should contains FilterPushDown", explainNotNull.contains(expectedFilterPushDownExplain));
+    List<Object[]> resultNotNull = sql(sqlNotNull);
+    assertEquals("should have 2 record", 2, resultNotNull.size());
+    List<Object[]> expected = Lists.newArrayList();
+    expected.add(new Object[] {1, "a"});
+    expected.add(new Object[] {2, "b"});
+    assertArrayEquals("Should produce the expected record", resultNotNull.toArray(), expected.toArray());
+
+    assertEquals("Should create only one scan", 1, scanEventCount);
+    Assert.assertEquals("Should push down expected filter", "data IS NOT NULL",
+        FlinkUtil.describe(lastScanEvent.filter()));
+  }
+
+  @Test
+  public void testFilterPushDownIsNull() {
+    String sqlNull = String.format("SELECT * FROM %s WHERE data IS  NULL", TABLE_NAME);
+    String explainNull = getTableEnv().explainSql(sqlNull);
+    assertTrue("explain should contains FilterPushDown", explainNull.contains(expectedFilterPushDownExplain));
+    List<Object[]> resultNull = sql(sqlNull);
+    assertEquals("should have 1 record", 1, resultNull.size());
+    assertArrayEquals("Should produce the expected record", resultNull.get(0), new Object[] {3, null});
+
+    assertEquals("Should create only one scan", 1, scanEventCount);
+    assertEquals("Should push down expected filter", "data IS NULL", FlinkUtil.describe(lastScanEvent.filter()));
+  }
+
+  @Test
+  public void testFilterPushDownNot() {
+    String sqlNot = String.format("SELECT * FROM %s WHERE NOT id = 1 ", TABLE_NAME);
+    String explainNot = getTableEnv().explainSql(sqlNot);
+    assertTrue("explain should contains FilterPushDown", explainNot.contains(expectedFilterPushDownExplain));
+    List<Object[]> resultNot = sql(sqlNot);
+    assertEquals("should have 2 record", 2, resultNot.size());
+    List<Object[]> expectedNot = Lists.newArrayList();
+    expectedNot.add(new Object[] {2, "b"});
+    expectedNot.add(new Object[] {3, null});
+    assertArrayEquals("Should produce the expected record", resultNot.toArray(), expectedNot.toArray());
+
+    assertEquals("Should create only one scan", 1, scanEventCount);
+    Assert.assertEquals("Should push down expected filter", "id != 1", FlinkUtil.describe(lastScanEvent.filter()));
+  }
+
+  @Test
+  public void testFilterPushDownBetween() {
+    String sqlBetween = String.format("SELECT * FROM %s WHERE id BETWEEN 1 AND 2 ", TABLE_NAME);
+    String explainBetween = getTableEnv().explainSql(sqlBetween);
+    assertTrue("explain should contains FilterPushDown", explainBetween.contains(expectedFilterPushDownExplain));
```

Comment:
How about changing this to check whether the `explainBetween` contains a more detailed string `FilterPushDown,the filters :ref(name="id") >= 1,ref(name="id") <= 2]]], fields=[id, data])` ?   `explainBetween.contains("FilterPushDown")` is not so accurate for me.

Reply:
I add the more detail string in `IN`,`BETWEEN`,`NO_BETWEEN` test case